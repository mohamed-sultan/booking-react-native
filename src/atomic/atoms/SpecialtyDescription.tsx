import React from 'react';
import {Text, StyleSheet, View} from 'react-native';
import TipIcon from './TipIcon';
import  { useThemeColors } from '@store/useTheme';

const SpecialtyDescription = ({title, description}: any) => {
  const colors = useThemeColors();

  const styles = getStyles(colors);
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        <TipIcon size={19} />
        <Text style={styles.title}>{title}</Text>
      </View>
      <Text style={styles.description}>{description}</Text>
    </View>
  );
};
export default React.memo(SpecialtyDescription);

const getStyles = (colors: any) => StyleSheet.create({
  container: {
    marginBottom: 24,
    backgroundColor: colors.LightBackground,
    paddingHorizontal: 13,
    borderRadius: 12,
    paddingVertical: 16,
    opacity: 0.7,
  },
  title: {
    fontSize: 15,
    fontWeight: '600',
    marginBottom: 8,
    color: colors.PrimaryText,
  },
  description: {
    fontSize: 14,
    color: colors.PrimaryText,
    lineHeight: 24,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

