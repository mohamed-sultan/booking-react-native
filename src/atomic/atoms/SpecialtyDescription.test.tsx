import React from 'react';
import renderer from 'react-test-renderer';
import SpecialtyDescription from './SpecialtyDescription'; // Adjust the path as needed

// Mock the theme hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    LightBackground: '#f5f5f5',
    PrimaryText: '#111111',
  }),
}));

// Mock the TipIcon component to simplify snapshot output
jest.mock('./TipIcon', () => {
  return ({ size }: any) => `MockedTipIcon size=${size}`;
});

describe('SpecialtyDescription', () => {
  it('renders correctly with title and description', () => {
    const tree = renderer
      .create(
        <SpecialtyDescription
          title="Sample Title"
          description="This is a sample specialty description."
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with an empty description', () => {
    const tree = renderer
      .create(
        <SpecialtyDescription
          title="Title Only"
          description=""
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
