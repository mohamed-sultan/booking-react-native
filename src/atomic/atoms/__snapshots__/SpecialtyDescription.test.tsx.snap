// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecialtyDescription renders correctly with an empty description 1`] = `null`;

exports[`SpecialtyDescription renders correctly with title and description 1`] = `null`;
